controller/metrics/metrics.go

```diff
@@ -17,6 +18,7 @@ import (
 type MetricsServer struct {
 	*http.Server
 	syncCounter        *prometheus.CounterVec
+	k8sRequestCounter  *prometheus.CounterVec
```

Comment:
Is this a new counter that shows the number of requests? Minor: could be a separate PR.

Reply:
Right. This is a counter which reports number of kubernetes api requests per application. Ideally, all information should be available in cache and counter should be zero. 

If for some reason cache don't have required information (e.g. application has resource which we cannot convert and have to load from k8s) then it will be easy to understand which app is causing performance degradation.